### PR TITLE
Add a new utility function to return the device info as a C string object.

### PR DESCRIPTION
**Pull request by diptorupd**
The PR adds a new helper function in dpctl_sycl_device_manager called `DPCTLDeviceMgr_GetDeviceInfoStr`. The function returns a C string containing the device attributes currently printed out by `DPCTLDeviceMgr_PrintDeviceInfo`. The change will unblock #542.

Added a helper to convert C++ strings to C strings, Reduces code duplication.

### CODE REVIEW

File: dpctl-capi/include/dpctl_sycl_device_manager.h [hunk 1]
```diff
@@ -76,6 +76,22 @@ DPCTL_API
 __dpctl_give DPCTLDeviceVectorRef
 DPCTLDeviceMgr_GetDevices(int device_identifier);
 
+/*!
+ * @brief Returns as a string a set of device info attributes.
```
Comment (oleksandr-pavlyk) [inline dpctl-capi/include/dpctl_sycl_device_manager.h:80]:
```suggestion
 * @brief Returns a set of device info attributes as a string.
```

File: dpctl-capi/source/dpctl_sycl_device_manager.cpp [hunk 2]
```diff
@@ -168,6 +168,32 @@ DPCTLDeviceMgr_GetDevices(int device_identifier)
...
+    if (D) {
+        try {
+            auto infostr = get_device_info_str(*D);
+            auto cstr_len = infostr.length() + 1;
+            cstr_info = new char[cstr_len];
+#ifdef _WIN32
+            strncpy_s(cstr_info, cstr_len, infostr.c_str(), cstr_len);
+#else
+            std::strncpy(cstr_info, infostr.c_str(), cstr_len);
+#endif
```
Comment (oleksandr-pavlyk) [inline dpctl-capi/source/dpctl_sycl_device_manager.cpp:185]: We have code snippets such as this sprinkled around the codebase. I think we should encapsulate them into a callable function and use that.

**Review by oleksandr-pavlyk — APPROVED** [review]
LGTM with small suggestions.

**Comment by oleksandr-pavlyk** [comment]
By the way, the Python API should be changed too, in `_sycl_device.pyx`

**Comment by diptorupd** [comment]
> 
> 
> By the way, the Python API should be changed too, in `_sycl_device.pyx`

Let us add that in #542

File: dpctl-capi/source/dpctl_sycl_device_manager.cpp (hunk 2 above)
Comment (diptorupd) [reply dpctl-capi/source/dpctl_sycl_device_manager.cpp:185]: agreed. I will make the change.

File: dpctl-capi/include/dpctl_sycl_device_manager.h (hunk 1 above)
Comment (diptorupd) [reply dpctl-capi/include/dpctl_sycl_device_manager.h]: Fixed.

File: dpctl-capi/source/dpctl_sycl_device_manager.cpp (hunk 2 above)
Comment (diptorupd) [reply dpctl-capi/source/dpctl_sycl_device_manager.cpp]: Done in a58afe9c11e2a9bf8352af85fbde1ce9502f8c2d

**Review by oleksandr-pavlyk — APPROVED** [review]
Looks good.

**Pull request merged**